### DISCUSSION TITLE
???

### DISCUSSION THREAD

**Discussion by natamartins** [community/community#60289] [Programming Help]
### Select Topic Area

Product Feedback

### Body

React.js or Vue.js ??

**Comment by ryuuzera** [comment]
the choice depends on your project requirements, team experience, and personal preferences. React.js is flexible and suitable for large-scale projects, while Vue.js is simpler and better for smaller projects.

React.js:
 - Offers a component-based architecture for reusability and modularity.
 - Great performance due to its virtual DOM implementation.
- Flexible and scalable, making it suitable for complex applications.

Vue.js:
- Known for its simplicity and ease of learning.
- Offers an opinionated structure for faster development.
- Cohesive ecosystem with official libraries and tools.

**Reply by Kashumy** [reply]
📜  a real boss writes xd assembler code. in the notebook
joking plain JS written in notepad is better than these two

🏋️‍♂️🏋️‍♂️🏋️‍♂️ be boss

**Comment by MUK-Dev** [comment]
Choosing a framework/library is like choosing your poison.

In general, whenever you are trying to pick a framework/library to learn you should consider the following things

1. Market competition
2. Market demand
3. Learning curve
4. Ecosystem

**1. Market Competition**
In this, you should consider how many people are available and actively looking for jobs or learning this library or framework.
In our case, React has a little higher competition than Vue.

**2. Market demand**
In this, you should consider how many jobs are available for this library or framework. This in my opinion is one of the most important points to consider when choosing a technology.
In our case, the difference between React and Vue jobs is night and day. React has been here longer and a lot of companies have adapted to it. So it has way more jobs than Vue.

**3. Learning curve**
I personally use React but I know people who are using Vue and they say a lot of good things about its developer experience and that it is easy to learn. React is a little difficult to learn when you start but after a while, you start to see the benefits of learning it.

**4. Ecosystem**
Both libraries have a good ecosystem. React would again win in this because it has been in use longer.

-------------

At the end of the day, you should try to become a better engineer, not the best developer. An engineer is technology independent but a developer is dependent on one technology. If you have your basics and fundamentals covered, it won't be a problem for you to adapt to another technology if the need arises.

Keep learning with consistent effort and you will be good at anything you do,
*Good Luck*

**Comment by ramamimu** [comment]
If you have strong foundation in javascript, you may will feel convinient to use React because its look like regular function. On the other hand, better to start with vue because it is more understandable structure (but i don't say that react > vue vice versa, both are the same good).

I use both and It is possible to learn both . I also completely agree if everything depend on what the market needed.

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @natamartins ! 

We’ve moved your post to our new Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.
